ggml-opencl.cpp

```diff
@@ -729,10 +905,11 @@ static cl_program build_program_from_source(cl_context ctx, cl_device_id dev, co
         exit(1);
     }
 
-    const char* compile_opts = "-cl-mad-enable -cl-unsafe-math-optimizations -cl-finite-math-only -cl-fast-relaxed-math "
-                               "-DQK4_0=32 -DQR4_0=2 -DQK4_1=32 -DQR4_1=2 -DQK5_0=32 -DQR5_0=2 -DQK5_1=32 -DQR5_1=2 -DQK8_0=32 -DQR8_0=1";
+    std::string compile_opts = "-cl-mad-enable -cl-unsafe-math-optimizations -cl-finite-math-only -cl-fast-relaxed-math "
+                               "-DQK4_0=32 -DQR4_0=2 -DQK4_1=32 -DQR4_1=2 -DQK5_0=32 -DQR5_0=2 -DQK5_1=32 -DQR5_1=2 -DQK8_0=32 -DQR8_0=1 "
```

Comment:
any reason we cannot put #define in the source?

Reply:
I believe it's because the kernel source is compiled from a string literal at runtime, and does not go through the preprocessor. 
@0cc4m any suggestions?

Reply:
We wanted to avoid the compiler warnings that come from preprocessor directives in a string, but we have them anyways now because of `K_QUANTS_PER_ITERATION` preprocess branching. I'm not sure what a good way of resolving this is. Can these warnings be disabled?